src/components/structures/TimelinePanel.tsx

```diff
@@ -1584,13 +1587,21 @@ class TimelinePanel extends React.Component<IProps, IState> {
             : null;
     }
 
+    /**
+     * Get a list of undecryptable events currently visible on-screen.
+     *
+     * @returns {MatrixEvent[] | null} A list of undecryptable events, or null if
+     *     the list of events could not be determined.
+     */
     public getVisibleDecryptionFailures(): MatrixEvent[] | null {
         const messagePanel = this.messagePanel.current;
         if (!messagePanel) return null;
 
         const messagePanelNode = ReactDOM.findDOMNode(messagePanel) as Element;
         if (!messagePanelNode) return null; // sometimes this happens for fresh rooms/post-sync
         const wrapperRect = messagePanelNode.getBoundingClientRect();
+        const screenTop = wrapperRect.top - VISIBLE_DECRYPTION_FAILURE_MARGIN;
+        const screenBottom = wrapperRect.bottom + VISIBLE_DECRYPTION_FAILURE_MARGIN;
```

Comment:
I'm afraid I didn't really follow your reasoning for adding a tolerance here. Can you explain?

Reply:
This has been updated to provide proper hysteresis.